lib/src/components/NxProgressBar/__tests__/NxProgressBar.test.tsx

```diff
@@ -4,191 +4,176 @@
+      const inProgressEl = renderEl({ labelSuccess: 'complete', value: 50 })!,
+          finishedEl = renderEl({ value: 100 })!,
+          finishedElWithProp = renderEl({ labelSuccess: 'complete', value: 100 })!,
+          finishedElWithPropAndMax = renderEl({ labelSuccess: 'complete', value: 10, max: 10 });
+
+      expect(inProgressEl).toHaveTextContent('current progress');
+      expect(finishedEl).toHaveTextContent('current progress');
+      expect(finishedElWithProp).not.toHaveTextContent('current progress');
+      expect(finishedElWithProp).toHaveTextContent('complete');
+      expect(finishedElWithPropAndMax).not.toHaveTextContent('current progress');
+      expect(finishedElWithPropAndMax).toHaveTextContent('complete');
     });
 
-    it('adds the appropriate modifier class to each step depending on whether it is above, below, or at the ' +
-         'current value', function() {
-      const zeroBar = getMountedComponent({ showSteps: true, max: 12, value: 0 }),
-          partialBar = getMountedComponent({ showSteps: true, max: 12, value: 3 }),
-          fullBar = getMountedComponent({ showSteps: true, max: 12, value: 12 });
+    it('is overriden by labelError when it is specified', function() {
+      const el = renderEl()!,
+          elWithError = renderEl({ labelError: 'oops!' })!;
 
-      expect(zeroBar.find('.nx-progress-bar__step--below-value')).not.toExist();
-      expect(zeroBar.find('.nx-progress-bar__step--at-value')).not.toExist();
-      expect(zeroBar.find('.nx-progress-bar__step--above-value').length).toBe(11);
+      expect(el).toHaveTextContent('current progress');
+      expect(elWithError).not.toHaveTextContent('current progress');
+      expect(elWithError).toHaveTextContent('oops!');
+    });
+  });
+
+  it('sets progress to 0 when labelError is defined', function() {
+    const elWithError = renderEl({ labelError: 'oops!', value: 50 })!;
+    expect(within(elWithError).getByRole('progressbar')).toHaveAttribute('value', '0');
+    expect(elWithError).toHaveTextContent('0%');
+  });
 
-      expect(partialBar.find('.nx-progress-bar__step--below-value').length).toBe(2);
-      expect(partialBar.find('.nx-progress-bar__step--at-value').length).toBe(1);
-      expect(partialBar.find('.nx-progress-bar__step--above-value').length).toBe(8);
+  it('assigns an aria-label when component\'s label text is not rendered', function() {
+    const progressBarWithLabel = quickRender().getByRole('progressbar');
+    const progressBarWithoutLabel = quickRender({ inlineCounter: true }).getByRole('progressbar');
 
-      // ensure correct ordering
-      expect(partialBar.find('.nx-progress-bar__step--at-value ~ .nx-progress-bar__step--below-value')).not.toExist();
-      expect(partialBar.find('.nx-progress-bar__step--above-value ~ .nx-progress-bar__step--at-value')).not.toExist();
+    expect(progressBarWithLabel).not.toHaveAttribute('aria-label', 'current progress');
```

Comment:
Related to my first comment, here I think `aria-label` itself is an implementation detail, and what we really want to test is that the accessible name comes out correct one way or the other.  I'd have the `toHaveAccessibleName` matcher for that.

Reply:
> Related to my first comment, here I think `aria-label` itself is an implementation detail, and what we really want to test is that the accessible name comes out correct one way or the other. I'd use the `toHaveAccessibleName` matcher for that.

Added here [b7b7f69](https://github.com/sonatype/sonatype-react-shared-components/pull/903/commits/b7b7f69d02d8dddb436e4c31c6197b5385a4e45a). Upon further testing, I realize that the accessible name changes as well if the counter is visible or not, so it tests the accessible name of: 1. with label + counter, 2. with label no counter, and 3. with counter no label. Not sure if this should be split up into two tests -